src/parser/Expression.ts

```diff
@@ -130,7 +130,7 @@ export class FunctionExpression extends Expression implements TypedefProvider {
         super();
         if (this.returnTypeToken) {
             this.returnType = util.tokenToBscType(this.returnTypeToken);
-        } else if (this.functionType.text.toLowerCase() === 'sub') {
+        } else if (this.functionType && this.functionType.text.toLowerCase() === 'sub') {
```

Comment:
Another case of "would previously throw an error in an edge case which will likely never happen"

Reply:
Fixed!